test/deploy-kind-cluster.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+reg_name='kind-registry'
+reg_port=$(echo $PRIVATE_REGISTRY | cut -d':' -f2)
+
+function deleteCluster() {
+  kind delete cluster --name=${CLUSTER_NAME}
+  if [ $? -ne 0 ]; then
+    echo "Unable to delete cluster - ${CLUSTER_NAME}"
+    return 1
+  fi
+
+  docker rm -f ${reg_name}  
+  if [ $? -ne 0 ]; then
+    echo "Unable to delete private registry - ${reg_name}"
+    return 1
+  fi
+  return 0
+}
+
+function createCluster() {
+  echo "Using private registry at ${PRIVATE_REGISTRY}"
+  # Deploy KIND cluster if not deploy
+  rc=$(which kind)
+  if [ -z "$rc" ]; then
+    echo "Kind is not installed or in the PATH. Installing..."
+    os=$(uname)
+    curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.7.0/kind-${os}-amd64
+    if [ $? -ne 0 ]; then
+     echo "Unable to download kind for ${os}."
+     return 1
+    fi
+    chmod +x ./kind
+    sudo mv kind /usr/local/bin
```

Comment:
Same as eksctl - not sure about installing these things via our scripts, especially the "sudo" part of it

Reply:
Move to install this in circle config.yml